frontend/src/shared/util/hooks.ts

```diff
@@ -28,18 +28,25 @@ export function useBackgroundRefresh (action: () => void, deps: readonly unknown
 
     const [isBackgroundRefreshing, setIsBackgroundRefreshing] = useState(false);
     const callbackAction = useCallback(action, [action, ...deps]);
-    
+
     useEffect(() => {
+        // Only start a periodic refresh if the condition is met
         if (condition) {
             const timerId = setInterval(() => {
+                // Once the condition is no longer met we can stop refreshing
+                if (!condition) {
+                    clearInterval(timerId);
+                    return;
+                }
                 setIsBackgroundRefreshing(true);
                 callbackAction();
             }, (window.env.BACKGROUND_REFRESH_INTERVAL || 60) * 1000);
-            
             return () => {
                 clearInterval(timerId);
             };
         }
-    }, [callbackAction, condition]);
+        // We only want to recreate the interval if the condition changes
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [condition]);
```

Comment:
why don't we want to recreate this hook when the function changes? what effect does it have when we have it there?

Reply:
Kyle can say for sure but I think it is "correct" to have callbackAction as a dependency. Likely the lambda being passed in counts as being new every render and we should really be doing use`useBackgroundRefresh(useCallback(...))`

Reply:
Since the component is remounting the callback was being "changed" which was triggering this `useEffect`. In practice that callback won't actually change (currently).